lib/tasks/maintenance.rake

```diff
@@ -43,4 +43,38 @@ namespace :maintenance do
       from_stage.destroy! if delete_empty && from_stage.deploy_groups.empty?
     end
   end
+
+  # FROM and TO handles multiple deploy groups to not mess with projects that have each deploy group in a single stage.
+  desc "delete deploy groups across stages for all projects. OWNER= DELETE=dg1,dg2 [DELETE_EMPTY=true]"
+  task delete_deploy_group: :environment do
+    projects = Project.where(owner: ENV.fetch("OWNER"))
+    delete = ENV.fetch("DELETE").split(",")
+    delete_group = DeployGroup.find_by_permalink!(delete)
+    delete_empty = (ENV["DELETE_EMPTY"] == "true")
+
+    puts "found #{projects.size} projects"
+
+    actions = projects.map do |project|
+      from_stage = project.stages.detect { |stage| delete & stage.deploy_groups.map(&:permalink) == delete }
+      [project, from_stage]
+    end
+
+    actions.select! do |project, from_stage|
+      if !from_stage
+        puts "#{project.name} Unable to find DELETE"
+      else
+        empty = (from_stage.deploy_groups == [delete_group])
+        puts "#{project.name} to be removed from:#{from_stage&.name}#{" (+ delete stage)" if empty}}"
```

Comment:
```suggestion
        puts "#{project.name} to be removed from:#{from_stage&.name}#{" (+ delete stage)" if empty && delete_empty}}"
```

Reply:
... doing the logic twice to decide if we should delete is kinda meh anyway, better to add a true/false flag to the `actions` ?